### PR TITLE
Add the cluster-autoscaler addon

### PR DESCRIPTION
**Pull request by xmudrii**
**What this PR does / why we need it**:

This PR adds the cluster-autoscaler addon.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of #3642

**Documentation**:

TBD once implemented in the UI.

**Does this PR introduce a user-facing change?**:

```release-note
Add the cluster-autoscaler addon
```

### CODE REVIEW

File: addons/cluster-autoscaler/cluster-autoscaler.yaml [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    matchLabels:
+      app: cluster-autoscaler
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: cluster-autoscaler
+    spec:
+      containers:
+      - image: k8s.gcr.io/autoscaling/cluster-autoscaler:${AUTOSCALER_VERSION}
```
Comment (xmudrii) [inline addons/cluster-autoscaler/cluster-autoscaler.yaml]: How can I pass some value (e.g. version in this case) to the addon manifest?

**Comment by toschneck** [comment]
@xmudrii it would be good also to document the usage of the addon.

**Review by xrstf — APPROVED** [review]
/approve

**Comment by xmudrii** [comment]
@toschneck This is only the initial PR to add the manifest. There are some changes required on the API/UI side, so we'll document it once everything is complete.

**Pull request merged**